src/Socket.c

```diff
@@ -747,6 +752,7 @@ int Socket_continueWrite(int socket)
 	}
 #endif
 
+	assert(5 >= pw->count);
```

Comment:
Here is another possible stack overwrite, if the caller sets pw->count > 5 !

Reply:
Again, this change will have no effect, that I can see.  (Apart from the assertion, but I don't think we want assertions in production code).

Reply:
Socket_writev() can return -22, and this was not handled here!

The assertions will be disabled, if compiled with macro NDEBUG defined. But for regression tests, this assertions may be useful. At least it is easier to debug a "crash" caused by an assertion, than some random behaviour caused by stack overwrite!

Reply:
I think it is correct, or at least my intention, that the execution path goes into the subsequent code block if the rc is -22 (interrupted).